libraries/HID/src/HID.cpp

```diff
@@ -64,12 +82,24 @@ int HID_::getDescriptor(USBSetup& setup)
 
 uint8_t HID_::getShortName(char *name)
 {
+    if(serial) {
+        byte seriallen=min(strlen_P(serial), ISERIAL_MAX_LEN);
+        for(byte i=0; i< seriallen; i++) {
+            name[i] = pgm_read_byte_near(serial + i);
+        }
```

Comment:
why don't you use the `strncat_P` function and implement the string coping yourself?

Reply:
strncat_P is for concatenating strings, i think in this case strcpy_P would work but it has no limit of bytes to be copied. In this case, the parameter is assumed to be stored in PROGMEM but all the string manipulations in USBAPI.h are written to support memory storage as well as PROGMEM so they are not using strxxx_P functions either. My option can be further extended to support both cases. I guess the way it is, its ok, not a big deal.

Reply:
I think using a existing function makes the intention of the code directly clear to the reader

Reply:
I'm not sure I understand how do you suggest to re-write this code? I'm happy to change it, please suggest. The Serial is good feature but much less important than GetFeature/SetFeature, this was the primary goal for this pull request.

Reply:
yes you are right, that this has no impact on the PR, it's just my personal opinion on the readability of code.